### ISSUE TITLE
TOTK: Please add an "Add Arrows" button

### ISSUE THREAD

**Issue by Skorfius** [marcrobledo/savegame-editors#308]
As tittle states, please add an add arrow/s button. Pretty simple and may think it unnecessary but it would tremendously help when starting fresh or simply to avoid the need to run around to find some arrows to then edit. Please and thank you!

**Comment by NateHevens** [comment]
Seconded

**Comment by marcrobledo** [comment]
This will be available in the next version!

**Comment by marcrobledo** [comment]
Added in 74cbcc8!

**Comment by Skorfius** [comment]
Thank you for adding this in, I had been anticipating it :)
However the arrows are unusable :(
I added arrows to a fresh save and as expected the edited amount of arrows shows up in the last slot of the bow tab as normal.
Although when you to attempt to fire an arrow, you cant and it'll give u a warning that you have no arrows along with the arrow count while aiming showing 0.
If you pick up any more arrows it'll stack in the inventory like normal but will still show as 0 when aiming down sights.

**Comment by marcrobledo** [comment]
Oh no, that's why I never added this feature: I feared it wouldn't work just by adding the item. There must be another flag that enables them. Here are my candidates:

```
0x59d8c4cd;Pouch.Arrow.IsValid #BoolArray
0x35ebdbb6;Pouch.Arrow.Combined.Life #Int32Array
0x53b27d94;Pouch.Arrow.Content.StockNum #Int32Array
0xd6c79114;Pouch.Arrow.EquipIndex #Int32Array
0x5fd21a91;Pouch.Arrow.Combined.Name #String64Array
0x24dd3262;Pouch.Arrow.Content.Name #String64Array

0x1ee3be73;IsGet.NormalArrow #Bool
0xa5adb894;IsGet.Obj_ArrowBundle_A_01 #Bool
0xcad06b9e;IsGet.Obj_ArrowBundle_A_02 #Bool

0x23112e94;IsGetAnyway.NormalArrow #Bool
0x9c38a014;IsGetAnyway.Obj_ArrowBundle_A_01 #Bool
0xd7922604;IsGetAnyway.Obj_ArrowBundle_A_02 #Bool
```

I'm pretty sure it is that `Pouch.Arrow.EquipIndex`, but I don't think the game allows you to equip arrows from the pause menu. I figure they get equipped automatically since this game has only an arrow slow (well, it has two internally, but I figure it's for an end game event).

Does the arrow icon in the pause menu appear with a black or a blue background? Can you post a raw savegame in the sky fields but without any arrow *never* obtained?

**Comment by Skorfius** [comment]
Sorry I took a small break off TOTK, ready to start all over again XD
Yes the arrow does appear with black background when adding with editor
Here's a really clean save pre arrows and after

[Arrows Fix.zip](https://github.com/marcrobledo/savegame-editors/files/11777711/Arrows.Fix.zip)

**Comment by tekacs** [comment]
Just tested this and was able to get the correct value due to @Skorfius' clean files, so thanks so much for uploading that!

It looks like the issue is indeed `Pouch.Arrow.EquipIndex` -- setting it to `0` fixes this issue. :)

**Comment by theslyskye** [comment]
Are we able to fix this yet? Or do I need to restart my game? lol

**Comment by inSaiyanxo** [comment]
> Just tested this and was able to get the correct value due to @Skorfius' clean files, so thanks so much for uploading that!
> 
> It looks like the issue is indeed `Pouch.Arrow.EquipIndex` -- setting it to `0` fixes this issue. :)

Thanks so much, your comment fixed the issue of arrows not working even though I have so many in the inventory.

**Comment by marcrobledo** [comment]
Fixed in a8585e4! Thank you for your contribution.

The last update comes with a big number of changes and fixes, so some features might be broken.
I'm doing an issue cleanup so I can start tracking the new issues easily.
This issue will be closed, but feel free to open a new one if the new update didn't fix your issue.
